cms/djangoapps/contentstore/signals.py

```diff
@@ -6,13 +7,29 @@
 
 from xmodule.modulestore.django import SignalHandler
 from contentstore.courseware_index import CoursewareSearchIndexer, LibrarySearchIndexer
+from contentstore.proctoring import register_proctored_exams
+from openedx.core.djangoapps.credit.signals import on_course_publish
 
 
 @receiver(SignalHandler.course_published)
 def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """
-    Receives signal and kicks off celery task to update search index
+    Receives publishing signal and performs publishing related workflows, such as
+    registering proctored exams, building up credit requirements, and performing
+    search indexing
     """
+
+    # first is to registered exams, the credit subsystem will assume that
+    # all proctored exams have already been registered, so we have to do that first
+    register_proctored_exams(course_key)
+
+    # then call into the credit subsystem (in /openedx/djangoapps/credit)
+    # to perform any 'on_publish' workflow
+    on_course_publish(course_key)
+
+    # Finally call into the course search subsystem
+    # to kick off an indexing action
+
     # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
     from .tasks import update_search_index
```

Comment:
Not part of this PR, but this import should probably be moved into the if block below